lib/cosmos/io/tcpip_server.rb

```diff
@@ -325,8 +358,18 @@ def start_listen_thread(port, listen_write = false, listen_read = false)
       end
     end
 
-    def listen_thread_body(listen_socket, listen_write = false, listen_read = false)
-      socket, address = listen_socket.accept()
```

Comment:
accept blocks forever.  Need to use accept_nonblock and the self-pipe trick to be able to shutdown the thread cleanly.